src/components/decimal/decimal.stories.tsx

```diff
@@ -1,21 +1,42 @@
 import React, { useState } from "react";
-import { ComponentStory, StoryFn } from "@storybook/react";
+import { ArgTypes, Meta, StoryObj } from "@storybook/react";
 
 import Decimal, { DecimalProps, CustomEvent } from ".";
 import CarbonProvider from "../carbon-provider/carbon-provider.component";
+import generateStyledSystemProps from "../../../.storybook/utils/styled-system-props";
 
-export const DefaultStory: ComponentStory<typeof Decimal> = (args) => {
-  const [state, setState] = useState("0.01");
-  const setValue = ({ target }: CustomEvent) => {
-    setState(target.value.rawValue);
-  };
+const styledSystemProps = generateStyledSystemProps({
+  margin: true,
+}) as Partial<ArgTypes<DecimalProps>>;
 
-  return (
-    <Decimal label="Decimal" value={state} onChange={setValue} {...args} />
-  );
+const meta: Meta<typeof Decimal> = {
+  title: "Decimal Input",
+  component: Decimal,
+  argTypes: {
+    ...styledSystemProps,
+  },
+};
+
+export default meta;
+type Story = StoryObj<typeof Decimal>;
+
+/* TODO: we really need a better of having a reusable default story that can show state
```

Comment:
**comment** (non-blocking): Yeah I agree with the comment, this sucks and doesn't even work properly (the code doesn't show correctly when you "show code", but then you're right that it didn't in the old storybook either). Since the shared part is so simple I think I'd just bite the bullet and rewrite each story separately with the controlled stuff just copied into each one - it's not that much code and I'm pretty sure this is what we do on other components.

Reply:
I decided against making this change for this PR as it's the same as it was before effectively and I think it'll ultimately be sorted by the ticket to review all of our stories